### ISSUE TITLE
support for JIRA integration?

### ISSUE THREAD

**Issue by c-mu** [serenity-bdd/serenity-cucumber#19]
Will there be support for the serenity-jira integration with cucumber? I'm currently using serenity-cucumber 1.1.1 and serenity-jira 1.0.1
I already tried all documented variations for the mapping of features/scenarios to stories "imported" from JIRA: 
* @issue resp. @Issue annotations on the feature like documented for jbehave, 
* @_jira issuekey_ annotation on the feature as recommended by cucumber, 
* mentioning the jira issuekey in the Feature title like documented for junit testclasses

but none of them map the acceptance tests to the appropriate JIRA stories.

Ideally this would work using the @_jira issuekey_ annotation since this is [the way recommended by cucumber](https://github.com/cucumber/cucumber/wiki/Tags) for mapping features and scenarios to requirements documented elsewhere. 
But I guess it may be easier to copy the handling of the @issue notation in the SerenityReporter class over from the jbehave plugin.

**Comment by YamStranger** [comment]
Hi @c-mu, Thanks for this issue. 

Please refer to updated manual [chapter](http://thucydides.info/docs/serenity-staging/#_integrating_with_jira) for manual about Jira integration: 


If you feel that this issue already solved - please close it.

Thanks a lot!

**Comment by c-mu** [comment]
Hi @YamStranger, thanks updating docs. I did the initial configuration of my project based on the blog post you used to update the manual.
As I stated above are using cucumber for writing the BDD tests. We annotated our tests like described for jbehave. Serenity picks up on the Jira settings and downloads issues, but it doesn't do so with the issue tags. As a result I get Jira issues in addition the features from the feature files in the report. 
I took a peek at the implementation. There seems to be some special treatment for the tag value "issue" in the reporter of jbehave, that I  could not find in the code of the cucumber plugin. This led me to the suspicion that the Jira integration may not be supported with cucumber yet.

**Comment by YamStranger** [comment]
@c-mu yes I see, thanks for report about this bug

**Comment by cliviu** [comment]
hi c-mu, I'm working on this

**Comment by YamStranger** [comment]
@c-mu It seems that now issue is fixed, can take a look on it please?

**Comment by YamStranger** [comment]
@c-mu It seems that this issue already fixed by @cliviu, I'll close this issue. 

Thanks!!!

**Comment by chainssethi** [comment]
How can we acheive this in serenity with cucumber. Can someone please give an example. I will really appreciate it.

**Comment by wakaleo** [comment]
Have you looked at http://thucydides.info/docs/serenity-staging/#_integrating_with_jira and tried adding the latest serenity-jira jars to your dependencies?

**Comment by chainssethi** [comment]
The example in the documentation is for Jbehave. I tried implementing the same with cucumber but wasn't able to. Is the implementation shown in the example in the serenity documentation applicable exactly for both jbehave and cucumber exactly or is there some difference?

**Comment by chainssethi** [comment]
And yes all the latest serenity-jira jars are being used

**Comment by YamStranger** [comment]
Hello @chainssethi 
Here is some project with issue tags. I is used old serenity-core but you can use it as example for new version of serenity
https://github.com/serenity-bdd/serenity-demos/tree/master/cucumber-webtests

**Comment by cliviu** [comment]
Hi @chainssethi ,
   Yes, the project mentioned by @YamStranger includes usages of the @issue tag which is working in the latest serenity version .
  Take a look for example direct at the https://github.com/serenity-bdd/serenity-demos/blob/master/cucumber-webtests/src/test/resources/features/search/display_product_details.feature

```java
@issue:ETSY-101
Feature: Display product details
```
you can also use the @issues annotation like this  

```java
@issues:ETSY-101,ETSY-102
```

You should also not forget about correctly configuring your JIRA connection in your serenity.properties file: (like mentioned by @wakaleo, is all described in http://thucydides.info/docs/serenity-staging/#_integrating_with_jira  )

```java
jira.url=
jira.project=
jira.username=
jira.password=
```

**Comment by chainssethi** [comment]
Thanks YamStranger ,cliviuand wakaleo. I was able to do this. I the Jira issue id on my scenario in my report.

**Comment by YamStranger** [comment]
@chainssethi 

If you feel that your issue is fixed - please close this ticket. 

Thanks for helping us make Serenity BDD better )

**Comment by charansethi1** [comment]
Hi. I have one more query. When i add issue tags to my feature or my scenarios in feature. I get jira links to my tests in the report which is what we desire. But In the report they also show up as Issues under tags in the report. Moreover, when i click on these issue tags to see the tests in them- I see the same test appearing twice. 

Has anyone seen this before or any idea about it?

**Comment by cliviu** [comment]
hi @charansethi1 , if you follow one of the issues under the  "Issues" links  you will see a report containing all tests annotated with that Issue, which I think it makes sense. Let me know if this  answers your question.

**Comment by charansethi1** [comment]
@cliviu Thanks for replying. The problem I see is that when i click on an issue under the "issues" I see the same test appear twice in the report which i feel is surely a bug.This happens for every test i put an issue tag for.

 Also We want to add a jira id for every scenario so We don't want the list of issues coming in the report. Is it possible to not have the list of issues to show under the "issues"?

**Comment by cliviu** [comment]
@charansethi1 , I will take a look and get back to you

**Comment by cliviu** [comment]
@charansethi1, I have found no way to avoid the list of issues to come in the report
It is true, it seems that the tests appear duplicated in that report . I will try to analyse this, to see why it happens

**Comment by cliviu** [comment]
hi @charansethi1, the duplicated tests issues problem is solved with the above commit.
There is also a possibility  to completely exclude the "Related tags" section from the report by setting the property : "show.related.tags=false" but I'm not sure if this what you need. For different problems please open a new issue

**Comment by jenny-zhen** [comment]
@YamStranger , In my JIRA system, I defined "Component/s", "Fix Version/s" and "Label" fields. At the same time, I created some stories with different value in these three fields. For example I defined story "A" with value "A-component",  "A-Version", "A-label"; defined story "B" with value   "B-component",  "B-Version", "B-label"; defined story "C", "D" and so on.  And I defined some stories with "None" value for these three fields. I defined "@issue:A" in the A feature file, and defined the test file for it. After running "mvn clean verify", I got the test report.  But in the stories tab, I only got the stories with "None" value in these three fields.  Can you tell me the reason? And if I want to list all the stories in the test report, can you give me some suggestion to fix it?

**Comment by jenny-zhen** [comment]
@YamStranger , @cliviu , @wakaleo , can someone help me to resolve my problem?  Thanks.

**Comment by cliviu** [comment]
Hi @jenny-zhen , can you provide us a small example to help us reproduce your problem ? Thanks !

**Comment by jenny-zhen** [comment]
@cliviu , thanks for your response.  Sorry I can not share the jira of my company. But I can show you the screenshots about my question: The attachment "report.png" is the test report screenshot and the JIRA.png is the screenshot of my JIRA with story filter.
![jira](https://cloud.githubusercontent.com/assets/19819954/16735193/38b145f0-4756-11e6-801b-4f05c7546df5.png)
![report](https://cloud.githubusercontent.com/assets/19819954/16735192/38ae9c9c-4756-11e6-8540-f91c1182124e.PNG)


And in my cucumber-serenity project, I only defined "@issue:RIVA-3162" in one feature file to connect the JIRA system.  Here is the feature file:

@issue:RIVA-3162
Feature:
  In order to quickly navigate back to some page
  As a user
  I would like to see a list of recent pages I have visited

  Scenario: visit some page
    Given user log in riva application correctly with username "<string>" and password "<string>"
    When user open "asset" list
    And user open the first asset page
    Then user get the asset page and the asset name

  Scenario: check the recent menu
    Given user reset the session
    When user navigate to the recent menu
    Then user will see a list of recent page including asset list and the first asset page
    When user click on the first asset page menu
    Then user quickly navigate back to the first asset page

And here is the definition about the JIRA in pom.xml:
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-jira-requirements-provider</artifactId>
            <version>1.1.1</version>
        </dependency>

        <plugin>
                <groupId>net.serenity-bdd.maven.plugins</groupId>
                <artifactId>serenity-maven-plugin</artifactId>
                <version>${serenity.version}</version>
                <executions>
                    <execution>
                        <id>serenity-reports</id>
                        <phase>post-integration-test</phase>
                        <goals>
                            <goal>aggregate</goal>
                        </goals>
                    </execution>
                </executions>
                <dependencies>
                    <dependency>
                        <groupId>net.serenity-bdd</groupId>
                        <artifactId>serenity-jira-requirements-provider</artifactId>
                        <version>1.1.1</version>
                    </dependency>
                </dependencies>
            </plugin>

My questions:
(1) Why in the "Stories" tab of the test report, I only get 5 story names?
(2) How can I list all the stories in the report? 
(3) And is there any method to allow me to list some kind of story? For example, I only want to list the stories with "closed" status? 

Thank you very much.

**Comment by jenny-zhen** [comment]
@cliviu , Just now I found the stories not listed in the stories tab do not define the  Epic link. That means the stories listed in the tab have the Epic link on the story definition page.  The RIVA-3177 and RIVA-3151 have the Epic link "New search list component".  And another three stories have the Epic link "System Admin Console".  Is it the reason? I mean every story has to link some Epic and then can be listed on the stories tab of the serenity report?

**Comment by cliviu** [comment]
hi @jenny-zhen , try to define the stories not associated with epics also in the feature file. You can associated more issues/stories with one feature, like @issues:RIVA-3159, RIVA-3161

**Comment by jenny-zhen** [comment]
@cliviu , Thanks a lot. I will try it. And here is the pom.xml. (I can not upload it as the xml format so I save it as docx. Maybe it is not easy to read. Sorry.)
[pom.docx](https://github.com/serenity-bdd/serenity-cucumber/files/359605/pom.docx)

**Comment by jenny-zhen** [comment]
@cliviu , I tried to define all the stories not associated with epics in the feature file now which looks lick below:
![feature1](https://cloud.githubusercontent.com/assets/19819954/16771689/6409269a-481f-11e6-84a6-6707c3ad43e6.PNG)

And after running the mvn verify, I got the report in which, the "Over Test Results" tab shows me the tests list with the stories I defined in the feature file (RIVA-3162,RIVA-3159,RIVA-3161,RIVA-3160), but the "Stories" tab still only show me the list in which stories not associated with epics.  

I am not sure this is designed or not? I mean whether only the stories associate with epics will be listed in the "Stories" tab or not?

Thank you.
